File: market-application/src/main/java/com/example/market/application/port/in/RefundBuyerUseCase.java
package com.example.market.application.port.in;

import com.example.market.domain.settlement.Refund;
import com.example.market.domain.trading.TradeId;

/**
 * InspectionFailed 이벤트 컨슈머. PG.refund() → Refund.complete → Trade.closeAsFailedAfterRefund.
 */
public interface RefundBuyerUseCase {
    Refund refund(TradeId tradeId);
}

File: market-application/src/main/kotlin/com/example/market/application/port/in/RefundBuyerUseCase.kt
package com.example.market.application.port.`in`

import com.example.market.domain.settlement.Refund
import com.example.market.domain.trading.TradeId

/**
 * InspectionFailed 이벤트 컨슈머. PG.refund() → Refund.complete → Trade.closeAsFailedAfterRefund.
 */
interface RefundBuyerUseCase {
    fun refund(tradeId: TradeId): Refund
}
